### PR TITLE
feat: examples

### PR DESCRIPTION
**Pull request by grolston**
*Issue #, if available:*

*Description of changes:*

Adds in example cloudformation templates and workflow templates for OIDC federation with IAM IdP setup. Example gives simple setup for either the OIDC IAM IdP setup only or creating the OIDC IAM IdP setup with a simple role. The role setup leverages specified GitHub Organization, Repository, and branch. The GitHub workflow example uses the OIDC role with managed policy to deploy a cloudformation template.

---

* [X] Have you followed the guidelines in our [Contributing guide?](https://github.com/aws-actions/configure-aws-credentials/blob/master/CONTRIBUTING.md)

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by kellertk — CHANGES_REQUESTED** [review]
Hi there! Thanks for the PR!

We definitely want to get some better examples for the action, but from experience customers will want to copy/paste examples if possible. I noticed we're referencing a `cloudformation/ec2-bastion.yml` file that's not present in the folder, so this isn't just a drop in and go example :)

Could we structure things this way instead?
```
examples/
    README.md
    <name of example1>/
        README.md
        cloudformation/
            <name of template>.yml
        workflows/
            <name of workflow>.yml
    <name of example2>/
        README.md
        cloudformation/
            <name of template>.yml
            <extra_file>.yml
        workflows/
            <name of workflow>.yml
    ...
```

That way each folder is a complete, self-contained example.

**Comment by grolston** [comment]
Pushing latest restructuring for the examples directory.

File: examples/cfn-deploy-example/README.md [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+
+Example uses aws-action `configure-aws-credentials` with OIDC federation. Prior to using this example project, the user needs to deploy the [github-actios-oidc-federation-and-role](../federated-setup/github-actions-oidc-federation-and-role.yml) template in the AWS account they want to deploy the CloudFormation template into. Specify the GitHub Organization name, repository name, and the specific branch you want to deploy on.
+
+Within the [github/workflows](./.github/workflows/) directory there is a [compliance.yml](./.github/workflows/compliance.yml) and a [deploy.yml](./.github/workflows/deploy.yml). The deploy.yml file leverages the aws-action `configure-aws-credentials` and accesses GitHub Action Secrets for some of the variables. The compliance.yml runs static application security testing using cfn-guard.
+
+To use the example you will need to set the following GitHub Action Secrets:
+
+| Secret Key | Used With | Description |
+| --------- | -------- | -----------|
+| AWS_ACCOUNT_ID | configure-aws-credential | The AWS account ID |
```
Comment (kellertk) [inline examples/cfn-deploy-example/README.md:11]:
```suggestion
| AWS_ACCOUNT_ID | configure-aws-credentials | The AWS account ID |
```

File: examples/cfn-deploy-example/README.md [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+| AWS_DEPLOY_ROLE | configure-aws-credential | The name of the IAM role |
```
Comment (kellertk) [inline examples/cfn-deploy-example/README.md:12]:
```suggestion
| AWS_DEPLOY_ROLE | configure-aws-credentials | The name of the IAM role |
```

File: examples/cfn-deploy-example/.github/workflows/compliance.yml [hunk 3]
```diff
@@ -0,0 +1,15 @@
+name: 'compliance'
+## run ci testing on all push events
+on: [push]
+jobs:
+  ## Guard rule set
+  sast-guard:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v2
```
Comment (kellertk) [inline examples/cfn-deploy-example/.github/workflows/compliance.yml:9]:
```suggestion
    - uses: actions/checkout@v3
```

File: examples/cfn-deploy-example/.github/workflows/deploy.yml [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+    runs-on: ubuntu-latest
+    # These permissions are needed to interact with GitHub’s OIDC Token endpoint.
+    permissions:
+      id-token: write
+      contents: read
+    steps:
+    - name: Checkout
+      uses: actions/checkout@v2
+    - name: Configure AWS Credentials
+      uses: aws-actions/configure-aws-credentials@v1
```
Comment (kellertk) [inline examples/cfn-deploy-example/.github/workflows/deploy.yml:23]:
```suggestion
      uses: aws-actions/configure-aws-credentials@v1-node16
```

**Review by kellertk — CHANGES_REQUESTED** [review]
Looks good! Just a few minor changes, please :)

File: examples/cfn-deploy-example/README.md (hunk 1 above)
Comment (grolston) [reply examples/cfn-deploy-example/README.md]: added fix

File: examples/cfn-deploy-example/.github/workflows/compliance.yml (hunk 3 above)
Comment (grolston) [reply examples/cfn-deploy-example/.github/workflows/compliance.yml]: changed version

File: examples/cfn-deploy-example/README.md (hunk 2 above)
Comment (grolston) [reply examples/cfn-deploy-example/README.md]: added fix

**Review by kellertk — APPROVED** [review]

**Pull request merged**